Mathlib/GroupTheory/Congruence/Basic.lean

```diff
@@ -181,6 +240,30 @@ noncomputable def quotientKerEquivOfSurjective (f : M →* P) (hf : Surjective f
     (ker f).Quotient ≃* P :=
   quotientKerEquivOfRightInverse _ _ hf.hasRightInverse.choose_spec
 
+/-- If e : M →* N is surjective then (c.comap e).Quotient ≃* c.Quotient with c : Con N -/
+@[to_additive "If e : M →* N is surjective then (c.comap e).Quotient ≃* c.Quotient with c :
+AddCon N"]
+noncomputable def comapQuotientEquivOfSurj (c : Con M) (f : N →* M) (hf : Function.Surjective f) :
+    (Con.comap f f.map_mul c).Quotient ≃* c.Quotient :=
+  (Con.congr Con.comap_eq).trans <| Con.quotientKerEquivOfSurjective
+  (c.mk'.comp f) <| Con.mk'_surjective.comp hf
+
+@[to_additive (attr := simp)]
+lemma comapQuotientEquivOfSurj_mk (c : Con M) {f : N →* M} (hf : Function.Surjective f) (x : N) :
+    comapQuotientEquivOfSurj c f hf x = f x := rfl
+
+@[to_additive (attr := simp)]
+lemma comapQuotientEquivOfSurj_symm_mk (c : Con M) {f : N →* M} (hf : Function.Surjective f)
+    (x : N) : (comapQuotientEquivOfSurj c f hf).symm (f x) = x :=
+  (MulEquiv.symm_apply_eq (c.comapQuotientEquivOfSurj f hf)).mpr rfl
+
+/-- This version infers the surjectivity of the function from a MulEquiv function -/
+@[to_additive (attr := simp)]
+lemma comapQuotientEquivOfSurj_symm_mk' (c : Con M) (f : N ≃* M) (x : N) :
+    ((@MulEquiv.symm (Con.Quotient (comap ⇑f _ c)) _ _ _
+    (comapQuotientEquivOfSurj c f f.surjective)) ⟦f x⟧) = ↑x :=
```

Comment:
```suggestion
      (comapQuotientEquivOfSurj c f f.surjective)) ⟦f x⟧) = ↑x :=
```

Reply:
Happy to make the change! Could you explain what's happening here so I know for the future? 

I'd thought hypotheses beyond the first line were indented twice (2 tabs, or 4 spaces), and then the proof itself begins at one indent (1 tab, or 2 spaces). 

Would this mean 
(A) first line is not indented, second line with hypotheses is indented twice, third line is indented 3 times, fourth 4 times, etc?

(B) first line is not indented, all further lines with hypotheses are indented twice, but this one gets an extra indent to show the grouping of the parentheses? (basically since we're splitting up the parenthesized ```
(@MulEquiv.symm (Con.Quotient (comap ⇑f _ c)) _ _ _
    (comapQuotientEquivOfSurj c f f.surjective))
```)

trying to make it multiple choice to reduce typing for you :)

Reply:
😁 thanks for the thought. Basically A, except that later lines can be less indented than what A says depending on hestructure of the expression